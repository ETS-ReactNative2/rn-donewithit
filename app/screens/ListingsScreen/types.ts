import { ImageSourcePropType } from 'react-native'

export type cardType = {
  title: string
  subTitle: string
  image: ImageSourcePropType
}
