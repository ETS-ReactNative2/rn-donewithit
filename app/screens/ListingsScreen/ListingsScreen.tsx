import { GestureResponderEvent, StyleSheet } from 'react-native'
import React, { useCallback, useMemo } from 'react'
import AppSafeAreaView from '../../components/AppSafeAreaView'
import { AppFlatList as List, AppCard as Card } from '../../components/lists'
import { cardType } from './types'
import { JACKET, COUCH } from '../../config/images'
import { LONG_TEXT } from '../../config/texts'
import styles from './styles'

export default function ListingsScreen() {
  const { cards } = useMemo(
    () => ({
      cards: [
        {
          title: LONG_TEXT,
          subTitle: LONG_TEXT,
          image: JACKET
        },
        {
          title: 'Couch in great condition',
          subTitle: '$1000',
          image: COUCH
        }
      ]
    }),
    []
  )
  const handlePress = useCallback((item: cardType) => {
    console.log(item)
  }, [])
  return (
    <AppSafeAreaView style={styles.container}>
      <List
        data={cards}
        keyExtractor={(item) => item.title}
        renderItem={({ item }) => {
          const { title, subTitle, image }: cardType = item
          return (
            <Card
              style={styles.card}
              title={title}
              subTitle={subTitle}
              image={image}
              onPress={(e?: GestureResponderEvent) => {
                handlePress(item)
              }}
            />
          )
        }}
      />
    </AppSafeAreaView>
  )
}
