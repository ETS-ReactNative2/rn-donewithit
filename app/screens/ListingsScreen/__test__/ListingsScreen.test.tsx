import React from 'react'
import renderer from 'react-test-renderer'
import ListingsScreen from '../index'

describe('ListingsScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListingsScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
