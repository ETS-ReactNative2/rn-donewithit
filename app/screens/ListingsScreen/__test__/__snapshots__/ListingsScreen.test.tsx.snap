// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingsScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#f8f4f4",
        "paddingBottom": 20,
      },
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "image": 1,
          "subTitle": "denouncing of a pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?
On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs to those who fail in their duty through weakness of will, which is the same as saying through shrinking from toil and pain. These cases are perfectly simple and easy to distinguish. In a free hour, when our power of choice is untrammeled and when nothing prevents our being able to do what we like best, every pleasure is to be welcomed and every pain avoided. But in certain circumstances and owing to the claims of duty or the obligations of business it will frequently occur that pleasures have to be repudiated and annoyances accepted. The wise man therefore always holds in these matters to this principle of selection: he rejects pleasures to secure other greater pleasures, or else he endures",
          "title": "denouncing of a pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?
On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs to those who fail in their duty through weakness of will, which is the same as saying through shrinking from toil and pain. These cases are perfectly simple and easy to distinguish. In a free hour, when our power of choice is untrammeled and when nothing prevents our being able to do what we like best, every pleasure is to be welcomed and every pain avoided. But in certain circumstances and owing to the claims of duty or the obligations of business it will frequently occur that pleasures have to be repudiated and annoyances accepted. The wise man therefore always holds in these matters to this principle of selection: he rejects pleasures to secure other greater pleasures, or else he endures",
        },
        Object {
          "image": 1,
          "subTitle": "$1000",
          "title": "Couch in great condition",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "marginHorizontal": 20,
                  "marginTop": 20,
                },
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "overflow": "hidden",
                },
              ]
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 200,
                  "width": "100%",
                }
              }
            />
            <View
              style={
                Object {
                  "paddingHorizontal": 20,
                  "paddingVertical": 15,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 32,
                      },
                      Object {
                        "color": "#000",
                      },
                    ],
                  ]
                }
              >
                denouncing of a pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?
On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs to those who fail in their duty through weakness of will, which is the same as saying through shrinking from toil and pain. These cases are perfectly simple and easy to distinguish. In a free hour, when our power of choice is untrammeled and when nothing prevents our being able to do what we like best, every pleasure is to be welcomed and every pain avoided. But in certain circumstances and owing to the claims of duty or the obligations of business it will frequently occur that pleasures have to be repudiated and annoyances accepted. The wise man therefore always holds in these matters to this principle of selection: he rejects pleasures to secure other greater pleasures, or else he endures
              </Text>
              <Text
                numberOfLines={3}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 32,
                      },
                      Object {
                        "color": "turquoise",
                      },
                    ],
                  ]
                }
              >
                denouncing of a pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?
On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs to those who fail in their duty through weakness of will, which is the same as saying through shrinking from toil and pain. These cases are perfectly simple and easy to distinguish. In a free hour, when our power of choice is untrammeled and when nothing prevents our being able to do what we like best, every pleasure is to be welcomed and every pain avoided. But in certain circumstances and owing to the claims of duty or the obligations of business it will frequently occur that pleasures have to be repudiated and annoyances accepted. The wise man therefore always holds in these matters to this principle of selection: he rejects pleasures to secure other greater pleasures, or else he endures
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#f8f4f4",
              "height": 1,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "marginHorizontal": 20,
                  "marginTop": 20,
                },
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "overflow": "hidden",
                },
              ]
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 200,
                  "width": "100%",
                }
              }
            />
            <View
              style={
                Object {
                  "paddingHorizontal": 20,
                  "paddingVertical": 15,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 32,
                      },
                      Object {
                        "color": "#000",
                      },
                    ],
                  ]
                }
              >
                Couch in great condition
              </Text>
              <Text
                numberOfLines={3}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Courier",
                      "fontSize": 13,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 32,
                      },
                      Object {
                        "color": "turquoise",
                      },
                    ],
                  ]
                }
              >
                $1000
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
